cpp/src/arrow/ArrowConfig.cmake.in

```diff
@@ -101,7 +101,7 @@ if(TARGET Arrow::arrow_static AND NOT TARGET Arrow::arrow_bundled_dependencies)
   foreach(CONFIGURATION ${arrow_static_configurations})
     string(TOUPPER "${CONFIGURATION}" CONFIGURATION)
     get_target_property(arrow_static_location Arrow::arrow_static
-                        LOCATION_${CONFIGURATION})
+                        IMPORTED_LOCATION_${CONFIGURATION})
```

Comment:
This is not a problem.
We can use `LOCATION_<CONFIG>` to get `IMPORTED_LOCATION` value:

https://cmake.org/cmake/help/latest/prop_tgt/IMPORTED_LOCATION.html

> To get the location of an imported target read one of the [LOCATION](https://cmake.org/cmake/help/latest/prop_tgt/LOCATION.html#prop_tgt:LOCATION) or LOCATION_<CONFIG> properties.

`LOCATION_<CONFIG>`: https://cmake.org/cmake/help/latest/prop_tgt/LOCATION_CONFIG.html

We should have set `IMPORTED_CONFIGURATIONS` to `Arrow::arrow_bundled_dependencies`:

```diff
diff --git a/cpp/src/arrow/ArrowConfig.cmake.in b/cpp/src/arrow/ArrowConfig.cmake.in
index 74e510114c..deb7bf94a8 100644
--- a/cpp/src/arrow/ArrowConfig.cmake.in
+++ b/cpp/src/arrow/ArrowConfig.cmake.in
@@ -103,6 +103,9 @@ if(TARGET Arrow::arrow_static AND NOT TARGET Arrow::arrow_bundled_dependencies)
     get_target_property(arrow_static_location Arrow::arrow_static
                         LOCATION_${CONFIGURATION})
     get_filename_component(arrow_lib_dir "${arrow_static_location}" DIRECTORY)
+    set_property(TARGET Arrow::arrow_bundled_dependencies
+                 APPEND
+                 PROPERTY IMPORTED_CONFIGURATIONS ${CONFIGURATION})
     set_target_properties(Arrow::arrow_bundled_dependencies
                           PROPERTIES IMPORTED_LOCATION_${CONFIGURATION}
                                      "${arrow_lib_dir}/${CMAKE_STATIC_LIBRARY_PREFIX}arrow_bundled_dependencies${CMAKE_STATIC_LIBRARY_SUFFIX}"
```

Reply:
> We can use LOCATION_<CONFIG> to get IMPORTED_LOCATION value:

I tested this (output in the issue) and it's not working, but neither is my fix so... :rofl:

Reply:
Do you have any preference between `LOCATION_<CONFIG>` and `IMPORTED_LOCATION_<CONFIG>`?
I prefer `LOCATION_<CONFIG>` to `IMPORTED_LOCATION_<CONFIG>` because https://cmake.org/cmake/help/latest/prop_tgt/IMPORTED_LOCATION.html mentions `LOCATION_<CONFIG>` not `IMPORTED_LOCATION_<CONFIG>`.
But it's not a strong opinion. If you prefer `LOCATION_<CONFIG>`, please revert this change before we merge this. If you prefer `IMPORTED_LOCATION_<CONFIG>`, we can merge this as-is.

Reply:
I don't have a strong opinion _but_ (nitpicking ahead) I am reading the cmake docs in a way that leads me to believe setting the `IMPORTED_LOCATION<CONFIG>` is the "right" thing:

[`LOCATION_<CONFIG>`](https://cmake.org/cmake/help/latest/prop_tgt/LOCATION_CONFIG.html):
> Read-only property providing a target location on disk.

IMO this mean it should not be set directly but rather using `IMPORTED_LOCATION_<CONFIG>` as the docs for [`IMPORTED_LOCATION`](https://cmake.org/cmake/help/latest/prop_tgt/IMPORTED_LOCATION.html) say:
> The IMPORTED_LOCATION target property may be overridden for a given configuration <CONFIG> by the configuration-specific [IMPORTED_LOCATION_\<CONFIG\>](https://cmake.org/cmake/help/latest/prop_tgt/IMPORTED_LOCATION_CONFIG.html#prop_tgt:IMPORTED_LOCATION_%3CCONFIG%3E) target property.

Reply:
Ah, wait. I want to say about `get_target_property(arrow_static_location Arrow::arrow_static LOCATION_${CONFIGURATION})` (read a property) not `set_target_property()`.